package/ota_daemon/src/ota_daemon.c

```diff
@@ -241,6 +241,11 @@ static int ota_version_check(const char *offered)
     return -1;
   }
 
+  if (offered_parsed.dev) {
```

Comment:
I think if we change this to only upgrade when the dev version is different we'll be good.